New module: CSS units and values

estelle opened the pull request:
CSS units and values module landing page.

Created a landing page for the module. [/en-US/docs/Web/CSS/CSS_Values_and_Units](https://pr38218.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/CSS_Values_and_Units) is basically a new page.

The two guides, [/en-US/docs/Web/CSS/CSS_Values_and_Units/numeric_data_types](https://pr38218.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/CSS_Values_and_Units/numeric_data_types) and
[/en-US/docs/Web/CSS/CSS_Values_and_Units/textual_data_types](https://pr38218.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/CSS_Values_and_Units/textual_data_types) were part of the landing page and have been moved to sub pages The content is not edited other than front matter, headers and links. So, they don't need a full review (unless you want to).

Other edits are links to both these pages, and a few titbits i noticed as i made those edits.

estelle committed:
New module: CSS units and values

estelle commented on files/en-us/learn_web_development/core/styling_basics/what_is_css/index.md:
```diff
@@ -89,7 +89,7 @@ h1 {
 - This example contains two declarations, one for `color` and another for `font-size`.
 
-Different CSS {{Glossary("property/CSS","properties")}} have different allowable values. In our example, we have the `color` property, which can take various [color values](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and_units#color). We also have the `font-size` property. This property can take various [size units](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and ...
+Different CSS {{Glossary("property/CSS","properties")}} have different allowable values. In our example, we have the `color` property, which can take various [color values](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and_units#color). We also have the `font-size` property. This property can take various [size units](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and ...
```
```suggestion
Different CSS {{Glossary("property/CSS","properties")}} have different allowable values. In our example, we have the `color` property, which can take various [color values](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and_units#color). We also have the `font-size` property. This property can take various [size units](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and_units#numbers_lengths_and_percentages) as a value.
```

estelle committed:
Update files/en-us/learn_web_development/core/styling_basics/what_is_css/index.md

estelle committed:
Merge branch 'main' into unitsguides

estelle committed:
New module: CSS units and values

estelle committed:
related concepts

estelle committed:
capitalization of links

chrisdavidmills reviewed (changes requested):
@estelle good work; few comments for you, but nothing too major.

estelle committed:
Update files/en-us/web/css/css_values_and_units/index.md

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

estelle commented on files/en-us/web/css/frequency-percentage/index.md:
```diff
@@ -68,7 +68,7 @@ No browser currently supports this feature.
 
 - [CSS data types](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_data_types)
-- [CSS Values and Units](/en-US/docs/Web/CSS/CSS_Values_and_Units)
+- [CSS values and units](/en-US/docs/Web/CSS/CSS_Values_and_Units) module
```
Per https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#link_text:

> Use sentence casing in the link text even if it is different from the linked page title or section title. It might be that the case used in the page or section title is incorrect. 

> For external links as well, use sentence casing even if the casing on the target article page is different. This is to ensure consistency across MDN Web Docs. Exceptions include names of books.

estelle replied:
The MDN module landing pages cover all the specs versions, hence "module" instead of "spec", and no numbers

estelle committed:
Merge branch 'main' into unitsguides

estelle committed:
updates per review

estelle commented:
part of https://github.com/openwebdocs/project/issues/218 and https://github.com/openwebdocs/project/issues/147

estelle commented:
part of https://github.com/openwebdocs/project/issues/220

estelle committed:
Merge branch 'main' into unitsguides

estelle commented on files/en-us/web/css/css_values_and_units/index.md:
```diff
@@ -1,331 +1,239 @@
 {{CSSRef}}
 
-Every CSS declaration includes a property / value pair. Depending on the property, the value can include a single integer or keyword, to a series of keywords and values with or without units. There are a common set of data types — values and units — that CSS properties accept. Below is an overview of most of these data types. Refer to the page for each value type for more detailed information.
+Every CSS declaration includes a property / value pair. Depending on the property, the value can include a single integer or keyword, to a series of keywords and values with or without units. Every property also accepts the CSS wides values. The CSS values and units module defines the data types — values and units — that CSS properties accept. This module also defines the CSS value definition syn ...
```
i added the global keywords to related concepts. In the linked page, where they were described, they're called "css-wide", not "global"; that section links out to the individual pages. The "key concepts" section in this page includes css-wide because it is listing the concepts listed in the module's multiple specs. I think we need to create a guide for those global values under cascade and inheritance where the values themselves are defined. In the textual data types guide that just provides the data type, we can use both terms: the official and the mdn used.

estelle commented on files/en-us/web/css/css_values_and_units/index.md:
```diff
@@ -1,331 +1,239 @@
+- [`vw`](/en-US/docs/Web/CSS/length#vw)
+- [`x`](/en-US/docs/Web/CSS/length#x)
+
+[Flex units](/en-US/docs/Web/CSS/CSS_Values_and_Units/numeric_data_types#flex_units) (`fr`) and [container units](/en-US/docs/Web/CSS/CSS_Values_and_Units/numeric_data_types#container_units) (`cqb`,`cqh`,`cqi`,`cqmax`,`cqmin`,`cqw`) are defined in the [CSS flexible box layout](/en-US/docs/Web/CSS/CSS_flexible_box_layout) and [CSS containment](/en-US/docs/Web/CSS/CSS_containment) modules, respecti ...
```
good catch.thanks.

estelle commented on files/en-us/web/css/css_values_and_units/index.md:
```diff
@@ -1,331 +1,239 @@
-- [CSS Basic Data Types](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_data_types)
-- [Learn: Values and units](/en-US/docs/Learn_web_development/Core/Styling_basics/Values_and_units)
-- [Trigonometric functions in CSS](https://web.dev/articles/css-trig-functions)
+- {{cssxref("initial")}}
```
moved to related concepts under cascade and inheritance grouping.

estelle commented on files/en-us/web/css/css_values_and_units/textual_data_types/index.md:
```diff
@@ -0,0 +1,110 @@
+
+{{CSSRef}}
+
+Every CSS declaration includes a property / value pair. Depending on the property, the value can include a single integer or keyword, to a series of keywords and values with or without units. There are a common set of data types — values and units — that CSS properties accept. Below is an overview of the textual data types. Refer to the page for each value type for more detailed information.
```
Intentionally included this intro sentence on each as the guides are not sequential; someone can land on any of these from multiple places

estelle commented on files/en-us/web/css/css_values_and_units/textual_data_types/index.md:
```diff
@@ -0,0 +1,110 @@
+
+Every CSS declaration includes a property / value pair. Depending on the property, the value can include a single integer or keyword, to a series of keywords and values with or without units. There are a common set of data types — values and units — that CSS properties accept. Below is an overview of the textual data types. Refer to the page for each value type for more detailed information.
+
+## Textual data types
```
agree. updated.

estelle commented on files/en-us/web/css/gradient/index.md:
```diff
@@ -189,5 +189,5 @@ span {
 - Gradient functions: {{cssxref("gradient/linear-gradient", "linear-gradient()")}}, {{cssxref("gradient/repeating-linear-gradient", "repeating-linear-gradient()")}}, {{cssxref("gradient/radial-gradient", "radial-gradient()")}}, {{cssxref("gradient/repeating-radial-gradient", "repeating-radial-gradient()")}}, {{cssxref("gradient/conic-gradient", "conic-gradient()")}}, {{cssxref("gradient/repeating ...
 - [CSS Basic Data Types](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_data_types)
-- [CSS Units and Values](/en-US/docs/Web/CSS/CSS_Values_and_Units)
+- [CSS units and values](/en-US/docs/Web/CSS/CSS_Values_and_Units) module
```
good catch

estelle committed:
edits per review

estelle commented on files/en-us/web/css/css_values_and_units/numeric_data_types/index.md:
```diff
@@ -0,0 +1,233 @@
+
+{{CSSRef}}
+
+Every CSS declaration includes a property / value pair. Depending on the property, the value can include a single integer or keyword, to a series of keywords and values, including numeric values with or without units. Numeric data types include {{cssxref("&lt;integer&gt;")}}, {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}}, and {{cssxref("&lt;percentage&gt;")}} values. This guide  ...
```
this is intentional, as most will not come from the module landing page.

estelle commented:
@chrisdavidmills Thanks for the excellent review. I think I've addressed everything. I'm leaving the intro sentence as most will not be comeing from landing page. otherwise, updated based on your other comments.

estelle committed:
syntax link

estelle committed:
syntax link

chrisdavidmills committed:
Update index.md

chrisdavidmills committed:
Update index.md

chrisdavidmills committed:
Update index.md

chrisdavidmills committed:
Update index.md

chrisdavidmills reviewed (approved):
Lovely, thanks @estelle. Your updates all look good; I'm approving and merging this now.

Full disclosure — I did a bit of messing around with the intro text for the numeric and textual data types pages, to use some of my text from the main values and units pages, but tailor it to be more about data types.

I thought you'd prefer me to get it done rather than go through another round of review, but let me know if you would like to see some more changes there, and I can open another PR.
